### ISSUE TITLE
Case studies demos don't work on iOS 13.2

### ISSUE THREAD

**Issue by nsillik** [pointfreeco/swift-composable-architecture#334]
**Describe the bug**
The case studies demos don't work on iOS 13.2. This issue is similar, if not identical, to #238.

This is the case in both the latest release (`0.9.0`) and the main branch as of today (at commit `c1f88bd8608e8a26fd61ab46c9f11a63b0f4023d`)

**To Reproduce**

- Clone the repository.
- Open `ComposableArchitecture.xcworkspace`
- Select the "Case Studies (SwiftUI)" scheme
- Select a simulator running 13.2 (I specifically tested this on the "iPhone 11 (13.2.2)" simulator)
- Build & Run
- In the application tap the "Basics" row
- Tap the "+" button on the counter
- Note the number doesn't change from zero

**Expected behavior**
The counter number should increase to 1

**Environment**
 - Xcode Version 12.3 (12C33)
 - Swift 5.3
 - OS (if applicable): iOS 13.2, 13.1, and 13.0

**Comment by RobertKoval** [comment]
iOS 13 itself has many bugs with SwiftUI and Combine until version 13.4

**Comment by nsillik** [comment]
Sure, but if you look at the other issues this was fixed and regressed. This is also not related to SwiftUI at all, it reproduces in the UIKit case studies as well. It's likely a combine issue, since it was fixed before, I'm wondering if there's some `#if available` trickery that would make it work across all iOS 13 versions. Otherwise I'm probably going to have to drop using TCA and I _really_ don't want to, I'm loving working with it.

**Comment by mluisbrown** [comment]
I have a suspicion that this issue might be related to this: https://forums.swift.org/t/question-about-valid-uses-of-observableobject-s-synthesized-objectwillchange/31141/2

In the ReactiveSwift fork of TCA, I added the following property in `ViewStore` to ensure that this wasn't a problem:

https://github.com/trading-point/reactiveswift-composable-architecture/blob/master/Sources/ComposableArchitecture/ViewStore.swift#L104

```swift
public lazy var objectWillChange = ObservableObjectPublisher()
```

In theory, just by conforming to `ObservableObject` (like `ViewStore` does), you get an `objectWillChange` property synthesised for you (some magic happening here to allow a protocol to have stored properties 🧙‍♂️), but if you don't have `@Published` property you get a new instance every time, which will cause the bug you are seeing. Note that `ViewStore` [no longer uses `@Published`](https://github.com/pointfreeco/swift-composable-architecture/pull/199) - also worth reading the comments on that PR btw.

It seems that in iOS 13.3 and above this issue was fixed in Combine.

So, I *think* this would be fixed it this property were added to `ViewStore`:

```swift
public lazy var objectWillChange = ObservableObjectPublisher()
```

It doesn't need `@available` checking as there's no harm in it being there even with the fixed versions of Combine.

@nsillik would you be able to test if the change above fixes this issue?

**Comment by nsillik** [comment]
@mluisbrown that's an awesome find! I'm in meetings all day today and tomorrow, but I'll check this and if it works submit a PR.

**Comment by Lixel** [comment]
@mluisbrown very nice. Seems to work! Thank you.

**Comment by stephencelis** [comment]
@nsillik Do you still plan on PRing a fix for this? Would love to have this working for folks 😄 If you don't have time we can look into it soon.

**Comment by nsillik** [comment]
Was just about to do this. Should have it done by the end of the day.

**Comment by nsillik** [comment]
@stephencelis  #353

**Comment by mluisbrown** [comment]
I guess this issue can be closed now that #353 was merged?
